OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -299,7 +300,8 @@ public bool TargetOverridesSelection(TargetModifiers modifiers)
 			public MoveDeployOrderTargeter(ActorInfo actorInfo, GrantConditionOnDeploy unit)
 			{
 				this.unit = unit;
-				info = actorInfo.TraitInfoOrDefault<MobileInfo>();
+				mobileInfo = actorInfo.TraitInfoOrDefault<MobileInfo>();
+				aircraftInfo = actorInfo.TraitInfoOrDefault<AircraftInfo>();
```

Comment:
Thinking out loud: we could add `MoveIntoShroud` and `CanEnterCell` (is this what the `MovementCostForCell` check was doing?) on the `IMoveInfo` interface, and then remove the explicit references to `MobileInfo` and `AircraftInfo` here and in Transforms.

Reply:
I think that would be an elegant solution. However, `CanEnterCell` is defined slightly differently for `AircraftInfo`. It checks whether the aircraft can land in the cell not whether it can move in the cell.

Reply:
I've implemented this using a new method `CanMoveInCell` to `IMoveInfo` that returns `CanEnterCell` in `Mobile` and always returns true in `Aircraft`.